src/transaction/components/TransactionProvider.tsx

```diff
@@ -62,13 +68,33 @@ export function TransactionProvider({
   }); // Component lifecycle
   const [transactionId, setTransactionId] = useState('');
   const [transactionHashList, setTransactionHashList] = useState<Address[]>([]);
+  const [transactions, setTransactions] = useState<
+    Call[] | ContractFunctionParameters[] | undefined
```

Comment:
When is `undefined`? Curios to understand this edge case.

Reply:
why does this need to be in state and/or could it just be initialized here instead of in a useEffect?

Reply:
good catch, initialized as a variable instead

I imagine there may be scenarios in the future where we want to allow developer overrides of `transactions` based on some state change (e.g. in `Pay`) where it needs to be a state, but made it a variable to be cleaner for now